ykrt/src/mt.rs

```diff
@@ -71,9 +70,11 @@ pub struct MT {
     job_queue: Arc<(Condvar, Mutex<VecDeque<Box<dyn FnOnce() + Send>>>)>,
     /// The hard cap on the number of worker threads.
     max_worker_threads: AtomicUsize,
-    /// How many worker threads are currently running. Note that this may temporarily be `>`
-    /// [`max_worker_threads`].
-    active_worker_threads: AtomicUsize,
+    /// [JoinHandle]s to each worker thread so that when an [MT] value is dropped, we can try
+    /// joining each worker thread and see if it caused an error or not. If it did,  we can
+    /// percolate the error upwards, making it more likely that the main thread exits with an
+    /// error. In other words, this [Vec] makes it harder for errors to be missed.
+    active_worker_threads: Mutex<Vec<JoinHandle<()>>>,
```

Comment:
Looks good, but one of the concerns I mentioned in the issue was the size of this vector being unbounded and consuming a lot of memory. Is it a real concern, do you think?

Reply:
No: it's bounded by `max_worker_threads`.

Reply:
Jolly good. Thanks.